### PR TITLE
Rewrite 'copy' functionality to remove jQuery dependency and follow GOV.UK Frontend conventions

### PR DESCRIPTION
**Pull request by igloosi**
- Removes jQuery dependency
- Followes GOV.UK Frontend convevtions
- Works in IE9+ (plugin does not support IE8)

Trello ticket: https://trello.com/c/J02wN45k/1326-update-copyjs-to-follow-frontend-conventions-and-remove-dependency-on-jquery

Ticket to build an ie8 bundle is here: https://trello.com/c/B6N3G2fS/1325-ensure-copy-to-clipboard-functionality-does-not-break-ie8-experience

### CODE REVIEW

File: src/javascripts/application.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ Example.init('.js-example__frame')
 Tabs.init('.js-example')
 
 // Add copy to clipboard to code blocks inside tab containers
-Copy.init('.app-tabs__container pre')
+new Copy('.app-tabs__container pre').init()
```
Comment (nickcolley) [inline src/javascripts/application.js]: I think to mirror GOV.UK Frontend the Copy constructor only accepts once instance of the place to enhance.

So we'd move the nodeListForEach outside into application.js.

It's also worth looking into if we could use `data-module="app-copy"`?

Comment (igloosi) [reply src/javascripts/application.js]: for the data- attribute i'd need to get #504 merged in so i can attach the attribute to it. i don't want to attach the attribute to each example container

**Comment by igloosi** [comment]
Now using `data-module`attribute instead of a css class to target copy functionality

File: src/javascripts/components/copy.js [hunk 2]
```diff
@@ -1,26 +1,41 @@
...
+  if (!$module) {
+    return
+  }
+  var $button = document.createElement('button')
+  $button.className = 'app-copy-button js-copy-button'
+  $button.setAttribute('aria-live', 'assertive')
+  $button.textContent = 'Copy'
+
+  var $parent = $module.parentNode
+  $parent.insertBefore($button, $parent.firstChild)
```
Comment (hannalaakso) [inline src/javascripts/components/copy.js]: Heads up for anyone else reviewing that https://github.com/alphagov/govuk-design-system/pull/518 will remove the need to look for the parent element here which is good as it feels a bit brittle (not the fault of this PR, this replicates the existing functionality).

**Comment by hannalaakso** [comment]
Merging in @igloosi's absence.

**Pull request merged**